tests/CMakeLists.txt

```diff
@@ -1,64 +1,27 @@
 # Authors: Giulio Romualdi
 # CopyPolicy: Released under the terms of the LGPLv2.1 or later
 
-cmake_minimum_required(VERSION 3.1)
-
-set (CMAKE_CXX_STANDARD 11)
-
-project(OsqpEigen-Test)
-
-if(NOT TARGET OsqpEigen::OsqpEigen)
-  find_package(osqp REQUIRED)
-  find_package(OsqpEigen REQUIRED)
-  find_package(Eigen3 REQUIRED)
-  include_directories(SYSTEM ${EIGEN3_INCLUDE_DIR})
-endif()
-
-find_package(GTest REQUIRED)
-include_directories(${GTEST_INCLUDE_DIRS})
-
-# adding support for checking the tests with valgrind
-if(RUN_VALGRIND_TESTS)
-     find_package(VALGRIND REQUIRED)
-     if(VALGRIND_FOUND)
-         set(CTEST_MEMORYCHECK_COMMAND ${VALGRIND_PROGRAM})
-         set(MEMORYCHECK_COMMAND ${VALGRIND_PROGRAM})
-         if (APPLE)
-             set(MEMORYCHECK_SUPPRESSIONS "--suppressions=${PROJECT_SOURCE_DIR}/cmake/valgrind-macos.supp")
-         else ()
-             set(MEMORYCHECK_SUPPRESSIONS "")
-         endif ()
-         set(MEMORYCHECK_COMMAND_OPTIONS "--leak-check=full --error-exitcode=1 ${MEMORYCHECK_SUPPRESSIONS}"  CACHE STRING "Options to pass to the memory checker")
-         mark_as_advanced(MEMORYCHECK_COMMAND_OPTIONS)
-         set(MEMCHECK_COMMAND_COMPLETE "${MEMORYCHECK_COMMAND} ${MEMORYCHECK_COMMAND_OPTIONS}")
-         separate_arguments(MEMCHECK_COMMAND_COMPLETE)
-     endif()
-endif()
-
-macro(add_osqpeigen_test classname)
-    set(testsrc ${classname}Test.cpp)
-    set(testbinary ${classname}Test)
-    set(testname   ${classname}Test)
-    add_executable(${testbinary} ${testsrc})
-    target_link_libraries(${testbinary} PRIVATE OsqpEigen::OsqpEigen osqp::osqp Eigen3::Eigen ${GTEST_LIBRARIES} pthread)
-    add_test(NAME ${testname} COMMAND ${testbinary})
-
-    if(RUN_VALGRIND_TESTS)
-        add_test(NAME memcheck_${testname} COMMAND ${MEMCHECK_COMMAND_COMPLETE} $<TARGET_FILE:${testbinary}>)
-    endif()
-endmacro()
-
-# QPTest
-add_osqpeigen_test(SparseMatrix)
-
-# QPTest
-add_osqpeigen_test(QP)
-
-# Update matrix
-add_osqpeigen_test(UpdateMatrices)
-
-# MPCTest
-add_osqpeigen_test(MPC)
-
-# MPCTest update matrix
-add_osqpeigen_test(MPCUpdateMatrices)
+add_osqpeigen_test(
+  NAME SparseMatrix
+  SOURCES SparseMatrixTest.cpp
+  LINKS OsqpEigen::OsqpEigen Eigen3::Eigen)
+
+add_osqpeigen_test(
+  NAME QP
+  SOURCES QPTest.cpp
+  LINKS OsqpEigen::OsqpEigen Eigen3::Eigen)
+
+add_osqpeigen_test(
+  NAME UpdateMatrices
+  SOURCES UpdateMatricesTest.cpp
+  LINKS OsqpEigen::OsqpEigen Eigen3::Eigen)
+
+add_osqpeigen_test(
+  NAME MPC
+  SOURCES MPCTest.cpp
+  LINKS OsqpEigen::OsqpEigen Eigen3::Eigen)
+
+add_osqpeigen_test(
+  NAME MPCUpdateMatrices
+  SOURCES MPCUpdateMatricesTest.cpp
+  LINKS OsqpEigen::OsqpEigen Eigen3::Eigen)
```

Comment:
Should the tests link ``Eigen3::Eigen``? Was there any issue in linking ``Eigen3::Eigen`` publicly in ``OsqpEigen``?

Reply:
I remember that back in time @traversaro suggested avoiding linking  `Eigen3::Eigen` publicly. Probably now something changed and we can link it

Reply:
Actually I think we can also remove `FindEigen3.cmake` file. Right?

Reply:
Yes, the problem with linking and installed target as a public dependency with `Eigen3::Eigen`  was that then you need  add `find_depedency(Eigen3 CONFIG REQUIRED)`  in your `OSQPEigen-config.cmake`  file. Back in time, we had to support platforms (i.e. the legacy Windows binaries) that contained Eigen but without distributing a `Eigen3Config.cmake`, so that line was not working out of the box. I think the problem is not there anymore, so we can just link as a public dependency  without problems.

Reply:
> 
> Actually I think we can also remove `FindEigen3.cmake` file. Right?

Yes!

Reply:
> Yes, the problem with linking and installed target as a public dependency with Eigen3::Eigen was that then you need add find_depedency(Eigen3 CONFIG REQUIRED) in your OSQPEigen-config.cmake file. Back in time, we had to support platforms (i.e. the legacy Windows binaries) that contained Eigen but without distributing a Eigen3Config.cmake, so that line was not working out of the box. I think the problem is not there anymore, so we can just link as a public dependency without problems.

Thanks! I did not remember that. I thought the problem was mainly in the inclusion in public headers.

Reply:
If we link ``Eigen`` as a public dependency, I guess it should be added also in https://github.com/robotology/osqp-eigen/blob/50154d5047a38b5b68b3a8994a6a1e76effabf36/CMakeLists.txt#L97
right?

Reply:
I think so

Reply:
In bc4b7845aa05a77e525d619a9e5d896cd3e4a62b `Eigen3` is now linked as `public` library